### PR TITLE
Make CQL and CRR compatible with Dataset Reader

### PR DESCRIPTION
**Pull request by avnishn**
First steps in removing the JSON Reader from RLlib

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kouroshHakha** [comment]
Have you looked at this PR? #25629

**Comment by gjoliver** [comment]
hey man, I am against this PR. For these offline methods, I don't think we should exhaustively read and add all the data in our RB. it won't work.
we should take the Marwil approach. also, the community PR that kourosh found is good.

**Comment by avnishn** [comment]
Ok so in response to this I'll do 2 things. 

1) shepherd the community PR in. 
2) Still remove the JSON reader as the default RLlib reader and then leverage the community PR

File: rllib/evaluation/worker_set.py [hunk 1]
```diff
@@ -599,8 +599,22 @@ def valid_module(class_path):
             )
         # JSON file or list of JSON files -> Use JsonReader (shuffled).
         else:
-            input_creator = lambda ioctx: ShuffledInput(
-                JsonReader(config["input"], ioctx), config["shuffle_buffer_size"]
+            _input = config["input"]
```
Comment (gjoliver) [inline rllib/evaluation/worker_set.py:602]: will it be easier if we just adapt the input into dataset reader format, so we can simply reuse the ``elif config["input"] == "dataset":`` block above?
and then we can just throw error msg here saying "unrecognized input format if we reach here".

**Pull request closed without merging**